packages/firebase_performance/lib/src/firebase_performance.dart

```diff
@@ -11,59 +11,82 @@ enum HttpMethod { Connect, Delete, Get, Head, Options, Patch, Post, Put, Trace }
 ///
 /// You can get an instance by calling [FirebasePerformance.instance].
 class FirebasePerformance {
-  FirebasePerformance._();
+  FirebasePerformance._(this._handle) {
+    channel.invokeMethod<bool>(
+      'FirebasePerformance#instance',
+      <String, dynamic>{'handle': _handle},
+    );
+  }
+
+  static int _nextHandle = 0;
 
-  static int _traceCount = 0;
-  static int _httpMetricCount = 0;
+  final int _handle;
```

Comment:
I do remember thinking about this. My goal was to try and separate the responsibility of the Plugin (`FirebasePerformancePlugin`) class from the main Firebase class (`FirebasePerformance`). My thinking would be that the plugin class would only be responsible for calling constructors and maintaining the `MethodCallHandlers`.

Also, i figured this wouldn't be as weird if it wasn't a Firebase plugin. Most libraries don't have a central class that most of their objects are created from. 

And this would also set us up to easily integrate multi-app support if the firebase team decides to add it!